inference-engine/ie_bridges/c/include/c_api/ie_c_api.h

```diff
@@ -371,6 +371,18 @@ INFERENCE_ENGINE_C_API(void) ie_core_versions_free(ie_core_versions_t *vers);
  */
 INFERENCE_ENGINE_C_API(IE_NODISCARD IEStatusCode) ie_core_read_network(ie_core_t *core, const char *xml, const char *weights_file, ie_network_t **network);
 
+/**
+ * @brief Reads the model from an xml string and a blob of the bin part of the IR. Use the ie_network_free() method to free memory.
+ * @ingroup Core
+ * @param core A pointer to ie_core_t instance.
+ * @param xml_content Xml content of the IR.
+ * @param weight_blob Blob containing the bin part of the IR.
+ * @param network A pointer to the newly created network.
+ * @return Status code of the operation: OK(0) for success.
+ */
+INFERENCE_ENGINE_C_API(IE_NODISCARD IEStatusCode) ie_core_read_network_from_memory(ie_core_t *core, const char *xml_content,
```

Comment:
I would afraid using null terminated strings in API due to possible security problems - why not using void* + byteSize.  Se how ie_memcpy declared. For fileNames i'm less concern. Also might be  having readCallback() is optimal way for extensibility and avoiding man in the middle attacks for encrypted weights files.